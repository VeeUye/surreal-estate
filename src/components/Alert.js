import React from "react";
import PropTypes from "prop-types";
import "../styles/alert.css";

const Alert = ({ message, success }) => {
  return (
    <div className={`alert alert--${success ? "success" : "error"}`}>
      {message}
    </div>
  );
};

Alert.propTypes = {
  message: PropTypes.string,
  success: PropTypes.bool,
}.isRequired;

export default Alert;
