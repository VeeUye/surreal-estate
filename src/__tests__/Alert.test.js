import React from "react";
import { render } from "@testing-library/react";
import Alert from "../components/Alert";

describe("Alert error", () => {
  const validProps = {
    message: "Error!",
  };
  it("renders correctly", () => {
    const { asFragment } = render(<Alert message={validProps.message} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("displays an error message", () => {
    const { getByText } = render(<Alert message={validProps.message} />);

    expect(getByText(/Error/).textContent).toBe("Error!");
  });
});

describe("Alert success", () => {
  const validProps = {
    message: "Success!",
  };
  it("renders correctly", () => {
    const { asFragment } = render(
      <Alert message={validProps.message} success />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("displays a success message", () => {
    const { getByText } = render(<Alert message={validProps.message} />);

    expect(getByText(/Success/).textContent).toBe("Success!");
  });
});
