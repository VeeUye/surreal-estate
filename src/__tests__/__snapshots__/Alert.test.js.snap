// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert error renders correctly 1`] = `
<DocumentFragment>
  <div
    class="alert"
  >
    Error!
  </div>
</DocumentFragment>
`;
